src/ORM/DataObjectSchema.php

```diff
@@ -494,6 +500,31 @@ public function compositeField($class, $field, $aggregated = true)
         return isset($fields[$field]) ? $fields[$field] : null;
     }
 
+    /**
+     * Returns a list of all the generated database columns on the class.
+     * Will check all applicable ancestor classes and aggregate results if $aggregated is true.
+     *
+     * @param string $class Name of class to check
+     * @param bool $aggregated Include fields in entire hierarchy, rather than just on this table
+     */
+    public function generatedFields(string $class, bool $aggregated = true): array
```

Comment:
Based on the `compositeFields()` method.
The `DataObjectSchema::cacheDatabaseFields()` method _always_ gets called during write so using this method instead of parsing `static::config()->get('db')` is faster.